libpod/define/annotations.go

```diff
@@ -145,6 +145,10 @@ const (
 	// of the init container.
 	InitContainerType = "io.podman.annotations.init.container.type"
 
+	// InfraNameAnnotation is used by generate and play kube when the infra container is set by the user during
+	// pod creation
+	InfraNameAnnotation = "io.podman.annotations.infra.name"
```

Comment:
> having the annotation would help translate a pod create --infra-name command to the generated kube yaml

I don't think we need the annotation.  By looking at the name of the pod's infra container, we know whether a custom name was used (i.e., when `name != $podID-infra`).

Reply:
> By looking at the name of the pod's infra container, we know whether a custom name was used (i.e., when name != $podID-infra).

But I guess then you'd not be able to replay the generated kube with the custom name because there seems to be no inbuilt `infraName` attribute. That's where this annotation comes in, even on regenerate, podman can effectively recreate the custom infra name.